DRAFT.md

```diff
@@ -1208,16 +1206,15 @@ constexpr polymorphic(allocator_arg_t, const Allocator& a,
 ```
 
 21. _Effects_: `allocator` is direct-non-list-initialized with `alloc`. If
-    `other` is valueless, `*this` is valueless. Otherwise, constructs an object
-    of type `polymorphic` that either owns the owned value of other, making
-    `other` valueless; or, owns an object of the same type constructed from
-    the owned value of `other` using the specified allocator, considering that
-    owned value as an rvalue.
-
-22. _[Note 1: This constructor does not require that `is_move_constructible_v<T>`
-  is `true`. --end note]_
-
-23. _[Note 2: The use of this function may require that `T` be a complete type
+    `other` is valueless, `*this` is valueless. Otherwise, if `alloc == other.alloc`
+    either constructs an object of type `polymorphic` that owns the owned value
+    of other, making`other` is valueless; or, owns an object of the same type
+    constructed from the owned value of `other` using the specified allocator,
+    considering that owned value as an rvalue. Otherwise constructs an object of
```

Comment:
I think this says the same thing twice...

> Otherwise constructs an object of type `polymorphic` using the specified allocator, considering that owned value as an rvalue.

How is this different from "owns an object of the same type constructed from the owned value of `other` using the specified allocator, considering that owned value as an rvalue." as I think this covers the case where allocators differ and the memory must be allocated from the new allocator and then constructed with `T`'s move constructor?

Reply:
Let's discuss offline

Reply:
Repetition is intended. Added a drafting note.